File: my-folder/0485-max-consecutive-ones/solution.c
int findMaxConsecutiveOnes(int* nums, int numsSize) {
    size_t result = 0, current = 0;
    for (size_t i = 0; i < numsSize; i++){
        if (nums[i] != 1){
            current = 0;
            continue;
        }
        current++;
        if (current > result)
            result = current;
    }
    return result;
}

File: my-folder/0485-max-consecutive-ones/solution.rs
impl Solution {
    pub fn find_max_consecutive_ones(nums: Vec<i32>) -> i32 {
        nums.split(|&num| num != 1)
        .map(|ones_sequence| ones_sequence.len() as i32)
        .max()
        .unwrap_or(0)
    }
}
